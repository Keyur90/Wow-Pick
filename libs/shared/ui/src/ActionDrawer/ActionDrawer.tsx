import React, { ReactNode } from 'react';
import Drawer from '@material-ui/core/Drawer';

interface ActionDrawerPropTypes {
  isOpen: boolean;
  onToggle: (force?: boolean) => void;
  children: NonNullable<ReactNode>;
}

const ActionDrawer: React.FC<ActionDrawerPropTypes> = ({ isOpen, onToggle, children }) => (
  <Drawer anchor={'bottom'} open={isOpen} onClose={() => onToggle(false)}>
    {children}
  </Drawer>
);

export { ActionDrawer };
