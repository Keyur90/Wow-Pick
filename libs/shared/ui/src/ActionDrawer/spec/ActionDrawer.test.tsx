import React from 'react';
import { render } from '@testing-library/react';
import { ActionDrawer } from '../ActionDrawer';

test('renders ActionDrawer', () => {
  const { container } = render(
    <ActionDrawer isOpen={true} onToggle={() => null}>
      Some actions
    </ActionDrawer>,
    {
      container: document.body,
    }
  );

  expect(container).toMatchSnapshot();
});
