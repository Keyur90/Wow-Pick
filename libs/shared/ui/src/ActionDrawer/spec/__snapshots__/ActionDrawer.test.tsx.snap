// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActionDrawer 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    class="MuiDrawer-root MuiDrawer-modal"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom MuiPaper-elevation16"
      tabindex="-1"
    >
      Some actions
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
